### DISCUSSION TITLE
Tenant creation without shared database

### DISCUSSION THREAD

**Discussion by pvaz** [abpframework/abp#7118] [Q&A]
Hi,

I did not found out the way to be able to use distinct databases for multiple tenants.

When I create a tenant I have the option after I create it to choose shared database or use another connection string.

But when I put a connection string that will use another database, it does not create the content if it does not exists.

How can I create the start content of this new database that will serve the tenant ?

If I create new features with tables/fields, how it will be maintained updated, tenants and default database ?

Thanks,
Paulo

**Comment by realLiangshiwei** [comment]
Hi,

You need to manually run the `.DbMigrator` application.

**Reply by pvaz** [reply]
Hi @realLiangshiwei ,

Understood, but I done that and the new database was not populated with any tables.

I'm using Blazor and version 4.1

Thanks,
Paulo

**Reply by realLiangshiwei** [reply]
Can you provide steps?

**Reply by pvaz** [reply]
Hi @realLiangshiwei,

Sure.

- I've created the Blazor solution, non tiered, Entity Framework Core, with a conntection string to my sql server main database that is the base database of the application.
- Using visual studio code, using integrated terminal, i changed to folder Application.DbMigrator and executed the command dotnet run.
- Database was populated correctly.
- changed to Application.HttpApi.Host and executed command dotnet run.
- On a new terminal window i changed to folder Application.Blazor and executed dotnet run.
- Everything is working well, I enter on the Blazor url app using the browser and do login without any problem.
- I go to Option Administration -> Saas -> Tenants -> Clicked on Option "New Tenant", filled with data: name, edition, email and password.
- Tenant is created and I can see it in the list of tenants. I choose Actions -> Connection Strings, remove the option "Use the Shared Database" and then I put the default connection string: "Server=PAULO\\SQLSERVER; Database=ApplicationDB2; Trusted_Connection=True;MultipleActiveResultSets=true; User Id=sa; Password=PaSsWoRd". ApplicationDB2 was already created mannualy before entering this option.
- Saved and in this stage it's logical that the Database should be populated with tables and some data, but does no happen.
- I go to dbmigrator, use donet run, nothing happens on new database.
- I go to compiled version, execute if, also nothing happens on new database.

Paulo

**Reply by mrs01dev** [reply]
Check the following code for creating tenants through migration ,it may help you.
[https://github.com/abpframework/abp/discussions/6909](url)
